src/BenchmarkDotNet/Toolchains/MonoWasm/WasmExecutor.cs

```diff
@@ -0,0 +1,146 @@
+                                      IDiagnoser diagnoser,
+                                      string executableName,
+                                      IResolver resolver)
+        {
+            ProcessStartInfo startInfo = WasmExecutor.BuildStartInfo(
+                CustomDotNetCliPath,
+                artifactsPaths.BinariesDirectoryPath,
+                artifactsPaths.ProgramName,
+                benchmarkId.ToArguments(),
+                redirectStandardInput: true);
+
+
+            startInfo.SetEnvironmentVariables(benchmarkCase, resolver);
+
+            using (var process = new Process { StartInfo = startInfo })
+            using (new ConsoleExitHandler(process, logger))
+            {
+                var loggerWithDiagnoser = new SynchronousProcessOutputLoggerWithDiagnoser(logger, process, diagnoser, benchmarkCase, benchmarkId);
+
+                logger.WriteLineInfo($"// Execute: {process.StartInfo.FileName} {process.StartInfo.Arguments} in {process.StartInfo.WorkingDirectory}");
+
+                diagnoser?.Handle(HostSignal.BeforeProcessStart, new DiagnoserActionParameters(process, benchmarkCase, benchmarkId));
+
+                process.Start();
+
+                process.EnsureHighPriority(logger);
+                if (benchmarkCase.Job.Environment.HasValue(EnvironmentMode.AffinityCharacteristic))
+                {
+                    process.TrySetAffinity(benchmarkCase.Job.Environment.Affinity, logger);
+                }
+
+                loggerWithDiagnoser.ProcessInput();
+                string standardError = process.StandardError.ReadToEnd();
+
+                process.WaitForExit(); // should we add timeout here?
+
+                if (process.ExitCode == 0)
+                {
+                    return new ExecuteResult(true, process.ExitCode, process.Id, loggerWithDiagnoser.LinesWithResults, loggerWithDiagnoser.LinesWithExtraOutput);
+                }
+
+                if (!string.IsNullOrEmpty(standardError))
+                {
+                    logger.WriteError(standardError);
+                }
+
+                return new ExecuteResult(true, process.ExitCode, process.Id, Array.Empty<string>(), Array.Empty<string>());
+            }
+        }
+
+
+        internal static ProcessStartInfo BuildStartInfo(string customDotNetCliPath,
+                                                        string workingDirectory,
+                                                        string programName,
+                                                        string benchmarkId,
+                                                        IReadOnlyList<EnvironmentVariable> environmentVariables = null,
+                                                        bool redirectStandardInput = false)
+        {
+            const string dotnetMultiLevelLookupEnvVarName = "DOTNET_MULTILEVEL_LOOKUP";
+
+            ProcessStartInfo startInfo = new ProcessStartInfo
+            {
+                FileName = "v8",
```

Comment:
`CommandLineOptions.WasmJavascriptEnginePath` allows for specifing a custom value, but anyway "v8" is always used here. We should either not allow for customization or use the provided value here